### PR TITLE
Add possibility to return all the FK frames in a vector

### PR DESCRIPTION
**Pull request by ahoarau**
Just saving intermediate results.
It's like a robot state publisher in a way.

### CODE REVIEW

File: orocos_kdl/src/chainfksolverpos_recursive.cpp [hunk 1]
```diff
@@ -56,7 +56,34 @@ namespace KDL {
             return 0;
         }
     }
+    int ChainFkSolverPos_recursive::JntToCart(const JntArray& q_in, std::vector<Frame>& p_out, int seg_nr)    {
+        unsigned int segmentNr;
```
Comment (smits) [inline orocos_kdl/src/chainfksolverpos_recursive.cpp]: You probably want to check that the p_out is of correct size!

File: orocos_kdl/src/chainfksolverpos_recursive.cpp [hunk 2]
```diff
@@ -56,7 +56,34 @@ namespace KDL {
...
+        if(seg_nr<0)
+            segmentNr=chain.getNrOfSegments();
+        else
+            segmentNr = seg_nr;
+
+        std::fill(p_out.begin(),p_out.end(),Frame::Identity());
```
Comment (smits) [inline orocos_kdl/src/chainfksolverpos_recursive.cpp]: Could you move the fill after line 74?

File: orocos_kdl/src/chainfksolvervel_recursive.cpp [hunk 3]
```diff
@@ -63,5 +63,38 @@ namespace KDL
...
+    
+    int ChainFkSolverVel_recursive::JntToCart(const JntArrayVel& in,std::vector<FrameVel>& out,int seg_nr)
+    {
+        unsigned int segmentNr;
+        if(seg_nr<0)
+            segmentNr=chain.getNrOfSegments();
+        else
+            segmentNr = seg_nr;
+
+        std::fill(out.begin(),out.end(),FrameVel::Identity());
```
Comment (smits) [inline orocos_kdl/src/chainfksolvervel_recursive.cpp]: Same here, move it after line 83

**Comment by smits** [comment]
minor comments, lgtm otherwise

**Comment by jf---** [comment]
Cool contribution...

**Comment by ahoarau** [comment]
The fill is moved. 

Do you think the size of the vector the total number of segments or just segmentsNr ?

https://github.com/ahoarau/orocos_kinematics_dynamics/blob/fk_solvers_all_seg_frames_ret/orocos_kdl/src/chainfksolverpos_recursive.cpp#L70

**Comment by smits** [comment]
I would say the size should be just segmentsNr.

**Comment by ahoarau** [comment]
Ok it's done, but no idea why travis failed on that last build.

**Pull request merged**